### PR TITLE
Adding export_as_csv method to usage license admin

### PR DESCRIPTION
**Pull request by ashleyelaine**

### CODE REVIEW

**Comment by theunraveler** [comment]
I wonder if we want _all_ of the model's fields in the export, or just the columns in the admin table view. Thoughts?

**Comment by ashleyelaine** [comment]
I was thinking about that, but then I thought maybe it's better to be able to select 'active' vs. 'inactive'? Or select specific ones? You're right though, we could just program it to download a CSV of all 'active' usage licenses. This would also make sense in case you weren't able to get all on one page in order to 'select all'. I did check on CC to see how pagination works, and it appears to be set up as a 'load more' so it would allow you to still select all.

**Comment by theunraveler** [comment]
Sorry, I think I wasn't clear. I do like the idea of selecting certain ones to export. What I meant was that it looks like the columns that get exported are _all of the model's fields_, where I think you might just want the columns from the admin table.

**Comment by ashleyelaine** [comment]
ohh I see :)

File: pyeti/eti_django/store/admin.py [hunk 1]
```diff
@@ -88,3 +90,20 @@ def sync_from_store(self, request, queryset):
         if failure:
             self.message_user(request, '%s license(s) failed to sync from store' % failure, level=messages.ERROR)
     sync_from_store.short_description = 'Sync from store'
+
+    def export_as_csv(self, request, queryset):
+
+        meta = self.model._meta
+        field_names = [field.name for field in meta.fields]
```
Comment (ashleyelaine) [inline pyeti/eti_django/store/admin.py]: @theunraveler so would you update `field_names` to be `field_names = ['token', 'num_seats', 'start_date', 'end_date', 'spree_order_number']` ?

Comment (theunraveler) [reply pyeti/eti_django/store/admin.py]: Yeah, I'd try interacting with the admin class to get those fields automatically. Sec and I'll send some code.

Comment (theunraveler) [reply pyeti/eti_django/store/admin.py]: Well, nevermind--involving the admin class to render the field values is a little harder than I thought. So yeah, I'd just update the `field_names` like you suggested.

**Comment by theunraveler** [comment]
I'm sorry---actually, here's what I'd do, so that it can be overridden in projects that subclass this class:

```
class UsageLicenseAdmin(admin.ModelAdmin):

    csv_export_fields = ['token', 'num_seats', 'start_date', 'end_date', 'spree_order_number']

    ...

    def export_as_csv(self, request, queryset):
        response = HttpResponse(content_type='text/csv')
        response['Content-Disposition'] = 'attachment; filename=%s.csv' % self.model._meta

        writer = csv.writer(response)
        writer.writerow(self.csv_export_fields)
        for obj in queryset:
            writer.writerow([getattr(obj, field) for field in self.csv_export_fields])

        return response

    export_as_csv.short_description = 'Export Selected'
```

That way, subclasses can just redefine `csv_export_fields` to change that settings, rather than overriding all of `export_as_csv`. Make sense?

**Comment by ashleyelaine** [comment]
That does make sense. I like it. Updated!

**Review by theunraveler — APPROVED** [review]
:+1:

**Pull request merged**